src/commands/publish/mod.rs

```diff
@@ -23,6 +23,36 @@ use crate::settings::project::{Project, ProjectType};
 
 pub fn publish(user: &GlobalUser, project: &Project, release: bool) -> Result<(), failure::Error> {
     info!("release = {}", release);
+
+    let missing_fields = match (project.account_id.is_empty(), project.name.is_empty()) {
+        (true, true) => Some("an account_id and name"),
+        (true, false) => Some("an account_id"),
+        (false, true) => Some("a name"),
+        (false, false) => None,
+    };
```

Comment:
Alternatively we could use a list to store all the missing fields and join them in the error message (omiting the `and` but I doesn't matter IMO).

Reply:
i think i would prefer this strategy as i think the code will be slightly more readable.